src/components/form/ContourForm.tsx

```diff
@@ -0,0 +1,242 @@
+
+            {formData?.conditional?.secureRoute && (
+              <>
+                <FormGroup
+                  className="pf-u-mt-md"
+                  fieldId="tls_termination"
+                  isRequired
+                >
+                  <CustomDropdown
+                    options={[
+                      { label: 'Edge', value: 'edge' },
+                      { label: 'Passthrough', value: 'passthrough' },
+                      { label: 'Re-encrypt', value: 're-encrypt' },
+                    ]}
+                    value={formData.conditional.termination}
+                    onChange={(value) => {
+                      updateFormData((prev) => ({
+                        ...prev,
+                        conditional: {
+                          ...prev.conditional,
+                          termination: value,
+                        },
+                      }));
+                    }}
+                    placeholder={t('select_termination')}
+                  />
+                  <div className="help-block">
+                    {t('route_tls_certificates')}
+                  </div>
+                </FormGroup>
+
+                <FormGroup className="pf-u-mt-md" fieldId="permit_insecure">
+                  <Switch
+                    id="permit-insecure"
+                    label={t('allow_insecure_traffic')}
+                    isChecked={formData?.conditional?.permitInsecure}
+                    onChange={(checked) =>
+                      updateFormData((prev) => ({
+                        ...prev,
+                        conditional: {
+                          ...prev.conditional,
+                          permitInsecure: checked,
+                        },
+                      }))
+                    }
+                  />
+                </FormGroup>
+
+                {formData?.conditional?.termination !== 'passthrough' && (
+                  <FormGroup
+                    className="pf-u-mt-md"
+                    fieldId="secret"
+                    label={t('secret')}
+                    isRequired
+                  >
+                    <CustomDropdown
+                      options={secrets.map((secret) => ({
+                        label: secret.metadata.name,
+                        value: secret.metadata.name,
+                      }))}
+                      value={formData?.conditional.secrets}
+                      onChange={(value) =>
+                        updateFormData((prev) => ({
+                          ...prev,
+                          conditional: {
+                            ...prev.conditional,
+                            secrets: value,
+                          },
+                        }))
+                      }
+                      placeholder={t('select_secret')}
+                    />
+                    <div className="help-block">{t('secret_info')}</div>
+                  </FormGroup>
+                )}
```

Comment:
Can we move it to another sub-component?
https://github.com/snapp-incubator/contour-console-plugin/pull/2#issuecomment-2591897860